src/main/java/com/codeborne/selenide/impl/WebdriverPhotographer.java

```diff
@@ -15,10 +18,42 @@ public class WebdriverPhotographer implements Photographer {
   @CheckReturnValue
   @Override
   public <T> Optional<T> takeScreenshot(Driver driver, OutputType<T> outputType) {
+    if (driver.config().fullPageScreenshots()) {
+      return takeFullScreenshot(driver, outputType);
+    } else {
+      return takeVisibleScreenshot(driver, outputType);
+    }
+  }
+
+  public <T> Optional<T> takeVisibleScreenshot(Driver driver, OutputType<T> outputType) {
     if (driver.getWebDriver() instanceof TakesScreenshot) {
       T screenshot = ((TakesScreenshot) driver.getWebDriver()).getScreenshotAs(outputType);
       return Optional.of(screenshot);
     }
     return Optional.empty();
   }
+
+  public <T> Optional<T> takeFullScreenshot(Driver driver, OutputType<T> outputType) {
+    if (driver.getWebDriver() instanceof ChromeDriver chromeDriver) {
+      long width = (long) chromeDriver.executeScript("return document.body.scrollWidth");
+      long height = (long) chromeDriver.executeScript("return document.body.scrollHeight");
+      long scale = (long) chromeDriver.executeScript("return window.devicePixelRatio");
+
+      HashMap<String, Object> setDeviceMetricsOverride = new HashMap<>();
+      setDeviceMetricsOverride.put("mobile", false);
+      setDeviceMetricsOverride.put("width", width);
+      setDeviceMetricsOverride.put("height", height);
+      setDeviceMetricsOverride.put("deviceScaleFactor", scale);
+      chromeDriver.executeCdpCommand("Emulation.setDeviceMetricsOverride", setDeviceMetricsOverride);
+
+      Map<String, Object> result = chromeDriver.executeCdpCommand("Page.captureScreenshot", new HashMap<>());
+
+      chromeDriver.executeCdpCommand("Emulation.clearDeviceMetricsOverride", new HashMap<>());
+
+      String base64 = (String) result.get("data");
+      T screenshot = outputType.convertFromBase64Png(base64);
+      return Optional.of(screenshot);
+    }
+    return Optional.empty();
```

Comment:
Not sure about behaviour in situation when this option is enabled but driver is not a chrome. Should we just return an empty value as well as selenium style screenshot or log/throw an error?